File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        l= 0
        for r in range(1, len(prices)):
            p = prices[r] - prices[l]
            profit = max(profit,p)
            if prices[r]<prices[l]:
                l=r
        return profit
            

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int l = 0;
        int ans = 0;
        for(int r=1; r<prices.length;r++){
            int profit = prices[r] - prices[l];
            ans = Math.max(ans, profit);
            if (prices[r]< prices[l]){
                l =r;
            }
        }
        return ans;
    }
}